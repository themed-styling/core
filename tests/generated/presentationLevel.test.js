import 'jest-styled-components'

import { presentationLevel } from '../../src/index.js'

import { testAllOn } from '../utilities.js'

testAllOn(presentationLevel, 'presentationLevel')
