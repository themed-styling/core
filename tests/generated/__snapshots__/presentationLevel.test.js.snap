// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`presentationLevel renders 0 1`] = `
Array [
  .c0 {
  presentation-level: 0;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: 0;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: 0;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: 0;
}

<div
    className="c0"
  />,
]
`;

exports[`presentationLevel renders function 1`] = `
Array [
  .c0 {
  presentation-level: hasnot;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: has;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: hasnot;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: has;
}

<div
    className="c0"
  />,
]
`;

exports[`presentationLevel renders invalid theme path 1`] = `
Array [
  .c0 {
  presentation-level: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: invalid.path.1;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: invalid.path.1;
}

<div
    className="c0"
  />,
]
`;

exports[`presentationLevel renders negative number 1`] = `
Array [
  .c0 {
  presentation-level: -54881;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: -54881;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: -42365;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: -42365;
}

<div
    className="c0"
  />,
]
`;

exports[`presentationLevel renders object [breakpoints + standard] 1`] = `
Array [
  .c0 {
  presentation-level: July;
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: September;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: November;
  }
}

@media screen and (min-width:August) {
  .c0 {
    presentation-level: August;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: July;
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: September;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: November;
  }
}

@media screen and (min-width:August) {
  .c0 {
    presentation-level: August;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: January;
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: December;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: April;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: May;
  }
}

@media screen and (min-width:August) {
  .c0 {
    presentation-level: June;
  }
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: January;
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: December;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: April;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: May;
  }
}

@media screen and (min-width:August) {
  .c0 {
    presentation-level: June;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`presentationLevel renders object [breakpoints] 1`] = `
Array [
  @media screen and (min-width:November) {
  .c0 {
    presentation-level: July;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: September;
  }
}

@media screen and (min-width:August) {
  .c0 {
    presentation-level: November;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {
  .c0 {
    presentation-level: July;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: August;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: September;
  }
}

@media screen and (min-width:August) {
  .c0 {
    presentation-level: November;
  }
}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: May;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: June;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: April;
  }
}

@media screen and (min-width:August) {
  .c0 {
    presentation-level: November;
  }
}

<div
    className="c0"
  />,
  @media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:November) {

}

@media screen and (min-width:August) {

}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: May;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: June;
  }
}

@media screen and (min-width:November) {
  .c0 {
    presentation-level: April;
  }
}

@media screen and (min-width:August) {
  .c0 {
    presentation-level: November;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`presentationLevel renders object [mediaQueries + standart] 1`] = `
Array [
  .c0 {
  presentation-level: July;
}

@media May {
  .c0 {
    presentation-level: August;
  }
}

@media April {
  .c0 {
    presentation-level: September;
  }
}

@media January {
  .c0 {
    presentation-level: November;
  }
}

@media April {
  .c0 {
    presentation-level: August;
  }
}

@media June {
  .c0 {
    presentation-level: November;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: July;
}

@media May {
  .c0 {
    presentation-level: August;
  }
}

@media April {
  .c0 {
    presentation-level: September;
  }
}

@media January {
  .c0 {
    presentation-level: November;
  }
}

@media April {
  .c0 {
    presentation-level: August;
  }
}

@media June {
  .c0 {
    presentation-level: November;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: April;
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    presentation-level: November;
  }
}

@media April {
  .c0 {
    presentation-level: January;
  }
}

@media January {
  .c0 {
    presentation-level: December;
  }
}

@media April {
  .c0 {
    presentation-level: April;
  }
}

@media June {
  .c0 {
    presentation-level: May;
  }
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: April;
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    presentation-level: November;
  }
}

@media April {
  .c0 {
    presentation-level: January;
  }
}

@media January {
  .c0 {
    presentation-level: December;
  }
}

@media April {
  .c0 {
    presentation-level: April;
  }
}

@media June {
  .c0 {
    presentation-level: May;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`presentationLevel renders object [mediaQueries] 1`] = `
Array [
  @media May {
  .c0 {
    presentation-level: July;
  }
}

@media April {
  .c0 {
    presentation-level: August;
  }
}

@media January {
  .c0 {
    presentation-level: September;
  }
}

@media April {
  .c0 {
    presentation-level: November;
  }
}

@media June {
  .c0 {
    presentation-level: August;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  @media May {
  .c0 {
    presentation-level: July;
  }
}

@media April {
  .c0 {
    presentation-level: August;
  }
}

@media January {
  .c0 {
    presentation-level: September;
  }
}

@media April {
  .c0 {
    presentation-level: November;
  }
}

@media June {
  .c0 {
    presentation-level: August;
  }
}

@media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

<div
    className="c0"
  />,
  @media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    presentation-level: January;
  }
}

@media April {
  .c0 {
    presentation-level: December;
  }
}

@media January {
  .c0 {
    presentation-level: April;
  }
}

@media April {
  .c0 {
    presentation-level: May;
  }
}

@media June {
  .c0 {
    presentation-level: June;
  }
}

<div
    className="c0"
  />,
  @media May {

}

@media April {

}

@media January {

}

@media April {

}

@media June {

}

@media May {
  .c0 {
    presentation-level: January;
  }
}

@media April {
  .c0 {
    presentation-level: December;
  }
}

@media January {
  .c0 {
    presentation-level: April;
  }
}

@media April {
  .c0 {
    presentation-level: May;
  }
}

@media June {
  .c0 {
    presentation-level: June;
  }
}

<div
    className="c0"
  />,
]
`;

exports[`presentationLevel renders postitive number 1`] = `
Array [
  .c0 {
  presentation-level: 54881;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: 54881;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: 42365;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: 42365;
}

<div
    className="c0"
  />,
]
`;

exports[`presentationLevel renders string 1`] = `
Array [
  .c0 {
  presentation-level: July;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: July;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: June;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: June;
}

<div
    className="c0"
  />,
]
`;

exports[`presentationLevel renders undefined 1`] = `
Array [
  <div
    className=""
  />,
  <div
    className=""
  />,
  <div
    className=""
  />,
  <div
    className=""
  />,
]
`;

exports[`presentationLevel renders valid theme path 1`] = `
Array [
  .c0 {
  presentation-level: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: DarkSlateBlue;
}

<div
    className="c0"
  />,
  .c0 {
  presentation-level: DarkSlateBlue;
}

<div
    className="c0"
  />,
]
`;
